src/Chapter2.hs

```diff
@@ -621,7 +647,15 @@ Write a function that takes elements of a list only in even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [Int] -> [Int]
+takeEven = go
+  where
+    go [] = []
+    -- Wrong even for positions not element ...
+    -- go (x:xs) = if even x then x : go xs else go xs
+    go [x]      = [x]
+    go (x:_:xs) = x : go xs
```

Comment:
Great pattern matching skills 💪🏼 

You can inline go function into `takeEven` too 🙂

Reply:
I will !